iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -38,91 +38,104 @@ internal sealed class HttpClientHelper : IHttpClientHelper
+                    _usingX509ClientCert = false;
                 }
 
-                (_httpClientHandler as WebRequestHandler).ClientCertificates.Add(clientCert);
-                _usingX509ClientCert = true;
-            }
+                if (proxy != DefaultWebProxySettings.Instance)
+                {
+                    webRequestHandler.UseProxy = proxy != null;
+                    webRequestHandler.Proxy = proxy;
+                }
 
-            if (proxy != DefaultWebProxySettings.Instance)
-            {
-                if (_httpClientHandler == null)
+                _httpMessageHandler = webRequestHandler;
+#elif NET5_0_OR_GREATER
+                var socketsHandler = new SocketsHttpHandler();
+                socketsHandler.SslOptions.EnabledSslProtocols = TlsVersions.Instance.Preferred;
+
+                if (!TlsVersions.Instance.CertificateRevocationCheck)
                 {
-                    _httpClientHandler = new WebRequestHandler();
+                    socketsHandler.SslOptions.CertificateRevocationCheckMode = X509RevocationMode.NoCheck;
                 }
 
-                _httpClientHandler.UseProxy = (proxy != null);
-                _httpClientHandler.Proxy = proxy;
-            }
+                if (transportSettings.ClientCertificate != null)
+                {
+                    socketsHandler.SslOptions.ClientCertificates.Add(transportSettings.ClientCertificate);
+                }
+                else
+                {
+                    _usingX509ClientCert = false;
+                }
+
+                if (proxy != DefaultWebProxySettings.Instance)
+                {
+                    socketsHandler.UseProxy = proxy != null;
+                    socketsHandler.Proxy = proxy;
+                }
+
+                _httpMessageHandler = socketsHandler;
 #else
+                var httpClientHandler = new HttpClientHandler();
+                httpClientHandler.SslProtocols = TlsVersions.Instance.Preferred;
+                httpClientHandler.CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck;
 
-            _httpClientHandler = httpClientHandler ?? new HttpClientHandler();
-            _httpClientHandler.SslProtocols = TlsVersions.Instance.Preferred;
-            _httpClientHandler.CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck;
+                if (transportSettings.ClientCertificate != null)
+                {
+                    httpClientHandler.ClientCertificates.Add(transportSettings.ClientCertificate);
+                    _usingX509ClientCert = true;
+                }
+                else
+                {
+                    _usingX509ClientCert = false;
+                }
 
-            if (clientCert != null)
-            {
-                _httpClientHandler.ClientCertificates.Add(clientCert);
-                _usingX509ClientCert = true;
-            }
+                if (proxy != DefaultWebProxySettings.Instance)
```

Comment:
Same as line 104?

Reply:
Oh, that is sockets handler.

Reply:
Yes, but I can't factor this if block out. The ```UseProxy``` and ```Proxy``` field are present in the implementation classes (HttpClientHandler, SocketsHttpHandler, WebRequestHandler) but are not present in the HttpMessageHandler interface.